### ISSUE TITLE
Consider moving initial submission uploads away from /tmp

### ISSUE THREAD

**Issue by emkll** [freedomofpress/securedrop#3067]
# Feature request

## Description

Submissions are initially streamed encrypted to disk in `/tmp`. If an adversary can execute code on the host, they can read and potentially alter the contents of the file.

This risk is mostly mitigated by the fact that files are encrypted and filenames are randomly generated and difficult to guess.

Ensure that the files are regularly cleared outside of the application context (Debian/Ubuntu by default clears `/tmp` on reboot) 

## User Stories

As a SecureDrop application, I do not want temporary files tampered by rogue processes on the host.

**Comment by legoktm** [comment]
I think this is fixed now, because the apache2 unit uses systemd's `PrivateTmp=true` feature. I uploaded a largeish file, and observed that the temporary aes file it was being streamed to ended up in `/tmp/systemd-private-{random thing}-apache2.service-{random thing}/tmp/{random thing}.aes`.

So systemd should prevent an attacker with code execution from tampering with files as they are uploaded, unless they gain execution as the www-data or root user. But in that case it's game over anyways.

@l3th3 can you confirm the above and that it's OK to close this?

**Comment by ghost** [comment]
Yes, this largely seems like a non-issue. `PrivateTmp=true` mitigates potential issues with regards to predictability, like symlink trickery, and the permissions are taken care of. And as you say, if they have execution with arbitrary access to the files, they can already do as much or worse than tamper with an encrypted file.
